app/src/main/java/org/feature/fox/coffee_counter/di/services/AchievementGeneration.kt

```diff
@@ -0,0 +1,164 @@
+package org.feature.fox.coffee_counter.di.services
+
+
+import org.feature.fox.coffee_counter.BuildConfig
+import org.feature.fox.coffee_counter.R
+import org.feature.fox.coffee_counter.data.local.database.tables.Achievement
+import org.feature.fox.coffee_counter.data.local.database.tables.Item
+import org.feature.fox.coffee_counter.data.local.database.tables.Purchase
+import org.feature.fox.coffee_counter.data.repository.UserRepository
+import org.feature.fox.coffee_counter.ui.common.showAchievementNotification
+import org.feature.fox.coffee_counter.util.UIText
+import java.text.SimpleDateFormat
+import java.time.LocalTime
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class AchievementGeneration @Inject constructor(
+    private val userRepository: UserRepository,
+    private val preference: AppPreference,
+) {
+
+    suspend fun checkAchievements(
+        items: List<Item>
+    ) {
+        val purchases: List<Purchase> =
+            userRepository.getPurchaseListOfUserDb(preference.getTag(BuildConfig.USER_ID))
+        val achievements: List<Achievement> =
+            userRepository.getAchievementListOfUserDb(preference.getTag(BuildConfig.USER_ID))
+        junkieAchievements(purchases, achievements, items)
+        nightPurchaseAchievement(purchases, achievements)
+        moneySpenderAchievement(purchases, achievements)
+    }
+
+    private suspend fun moneySpenderAchievement(
```

Comment:
I think the totalSpentMoney could be refactored, so it's calculated at the beginning of the method at least one of the 3 achievements isn't unlocked yet.